src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryFileValidator.cs

```diff
@@ -0,0 +1,172 @@
+                            string folderPath = PathWrapper.GetDirectoryName(filePath);
+                            if (isLocal)
+                            {
+                                if (!string.IsNullOrEmpty(folderPath) && !Directory.Exists(folderPath))
+                                {
+                                    errorMessage = string.Format(SR.InvalidBackupPathError, folderPath);
+                                    result = false;
+                                    break;
+                                }
+                            }
+                            else
+                            {
+                                bool isFolderOnRemote = true;
+                                bool existsOnRemote = IsPathExisting(serverConnection, folderPath, ref isFolderOnRemote);
+                                if (!existsOnRemote)
+                                {
+                                    errorMessage = string.Format(SR.InvalidBackupPathError, folderPath);
+                                    result = false;
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    result = false;
+                }
+            }
+            else
+            {
+                result = false;
+            }
+
+            return result;
+        }
+
+        #region private methods
+
+        private static bool IsPathExisting(ServerConnection serverConnection, string path, ref bool isFolder)
+        {
+            Enumerator en = null;
+            DataSet ds = new DataSet();
+            ds.Locale = CultureInfo.InvariantCulture;
+            Request req = new Request();
+            en = new Enumerator();
+            bool isExisting = false;
+            isFolder = false;
+
+            try
+            {
+                req.Urn = "Server/File[@FullName='" + Urn.EscapeString(path) + "']";
+                ds = en.Process(serverConnection, req);
+                if (ds.Tables != null && ds.Tables.Count > 0 && ds.Tables[0].Rows != null && ds.Tables[0].Rows.Count > 0)
+                {
+                    isFolder = !(Convert.ToBoolean(ds.Tables[0].Rows[0]["IsFile"], CultureInfo.InvariantCulture));
+                    isExisting = true;
+                }
+            }
+            finally
+            {
+                ds.Dispose();
+            }
+
+            return isExisting;
+        }
+
+        private static string GetMachineName(string sqlServerName)
+        {
+            string machineName = "";
+            if (sqlServerName != null)
+            {
+                // special case (local) which is accepted SQL(MDAC) but by OS
+                if ((sqlServerName.ToLowerInvariant().Trim() == localSqlServer) || (sqlServerName.ToLowerInvariant().Trim() == localMachineName))
```

Comment:
>sqlServerName.ToLowerInvariant().Trim() [](start = 21, length = 39)

if you save this value into a variable you don't need to call the exact same functions twice in a row..probably makes the if clause more readable too.